File: Practice/unix 3, 37/Завдання 3/python/source.py
#!/usr/bin/python3
import glob
import shutil
import pathlib
import sys
import os

if(sys.argv[1] == "--help"):
	print("Usage: source.py  DIR DIR_TO SIZE N")
	print("Copy first N files exceeding SIZE from DIR to DIR_TO")
	print("-h, --help show info about program")
	print("-t, --test start program with same parameters")
elif(sys.argv[1] == "--test"):
	os.system("./test.bash")
elif(len(sys.argv) != 5):
	print("Usage: source.py --help for more information")
elif(len(sys.argv) == 5):
	dir_from = sys.argv[1]
	dir_to = sys.argv[2]
	size_MAX = int(sys.argv[3])
	counter_MAX = int(sys.argv[4])
	current_size = 0
	counter = 0
	list_of_files = os.listdir(dir_from)
	list_of_files.sort(key=lambda f: os.stat(os.path.join(dir_from, f)).st_size)
	print(list_of_files)
	for path, subdirs, files in os.walk(dir_from):
		for item in list_of_files:
			for file in files:
				if file == item:
					if current_size < size_MAX and counter < int(counter_MAX):
						current_size = current_size + int(os.stat(os.path.join(dir_from, file)).st_size)
						shutil.move(os.path.join(dir_from, item), dir_to)
						counter = counter + 1

File: Practice/unix 3, 37/Завдання 3/perl/source.pl
#!/usr/bin/perl
if ($ARGV[0] eq "--help")
{
        print "Usage: source.pl  DIR DIR_TO SIZE N";
        print "Move first N files exceeding SIZE from DIR to DIR_TO\n";
        print "-h, --help show info about program\n";
        print "-t, --test start program with same parameters\n";			
}
elsif ($ARGV[0] eq "--test")
{
        system("./test.bash");
}
elsif (scalar @ARGV != 4)
{
        print "source.pl: missing file operands\n";
        print "Try 'source.pl --help' for more information\n";
}
else
{
    use File::Find::Rule;
    use File::Copy;
    use File::Basename;
	use strict;
	use warnings;
	use 5.010;
 
    my @files = File::Find::Rule->file()
                                ->in($ARGV[0]);
	my $total_size = 0;
	my $count = 0;
	my @sort_files = sort { $a cmp $b } @files;
        for my $file (@sort_files)
        {
		my $size = -s $file;
		$count = $count + 1;
		$total_size = $total_size + $size;
		if ($total_size < $ARGV[2] && $count <= $ARGV[3])
		{
			move($file, $ARGV[1]) or die "Unable to move: $!";
		} 
	}
        
}

